Rename kernel reserved function names

pfl opened the pull request:
This patch uses the scripts created in #12086 to rename reserved function names in the kernel/ directory and applying the changes found to the rest of the source code tree. The prefixes that are left as-is are '_Pragma', '_pragma', '__atomic', '__builtin' and '__SYSCALL', where the last one is left alone only due to practical reasons, there is already a macro '_SYSCALL' that gets renamed to 'Z_SYSCALL'.

The .c and .h files are updated, and the code compiles, but some of the linking is not yet detected, so this isn't fully working yet. An attempt to hack up on the syscall generating script has also been done.

This patch set is part of #9882.

pfl committed:
kernel/sched: Rename scheduler spinlock

Rename scheduler spinlock sched_lock to sched_spinlock as it will
collide with the cleanup of the reserved function name _sched_lock(),
which will also be called sched_lock().

Signed-off-by: Patrik Flykt <patrik.flykt@intel.com>

andyross reviewed (dismissed):
No complaints in general.   One thing that needs to be fixed is that (if I'm understanding it right) the series is not bisectable.  The build breaks between the second patch (which renames the APIs) and the third (which changes usage).  We really need those to be simultaneous.  If the change becomes too large for a single patch, you can break it up by API instead of directory.

A few notes on the naming chosen:

* Some of our more oddball naming should be fixed manually, I think.  "z_Cstart()" and "Swap()" aren't very good names (and weren't before).  I'd suggest z_swap() and z_cstart().

* Naming that has existing prefixes should probably combine that with the "z-ification" instead of just making the prefix longer.  The "_impl_" prefix is already applied to a symbol name (e.g. k_frog_dissect()) that is namespaced, so I'd argue we don't want to spend the bytes on a separate "z_" and should just call it "impl_*()".

* Similarly "z_arch_" could just be "zarch_" and save a byte.

pfl commented:
Yes, these should be squashed together, bisectability is quite dead in between. It is in two parts now so that one has some chance to figure out which output comes from which step, should someone be interested in dissecting brokenness. The naming can be changed, the scripts work decently well so they can be updated and the procedure re-applied. Re-running the scripts needs to happen already now, there have been some changes in the source tree already causing conflicts.

The naming and prefixes could be discussed for example in #9882.

andrewboie commented on arch/arc/core/prep_c.c:
```diff
@@ -119,8 +119,8 @@ void _PrepC(void)
-	_data_copy();
-	_Cstart();
+	z_bss_zero();
+	data_copy();
```
z_data_copy()?

pfl replied:
data_copy() comes from the previous kernel patch in file kernel/include/kernel_internal.h:
```
 #ifdef CONFIG_XIP
-void _data_copy(void);
+void data_copy(void);
 #else
-static inline void _data_copy(void)
+static inline void data_copy(void)
```
Coccinelle sees the static inline void _data_copy() and correctly renames it data_copy(). Of course the above wants it the opposite way.

pfl replied:
Fixed.

andrewboie commented on arch/arc/core/thread.c:
```diff
@@ -131,7 +131,7 @@ void _new_thread(struct k_thread *thread, k_thread_stack_t *stack,
 	}
 
-	_new_thread_init(thread, pStackMem, stackSize, priority, options);
+	new_thread_init(thread, pStackMem, stackSize, priority, options);
```
z_new_thread_init

pfl replied:
in kernel/device.c we have:
```
-static ALWAYS_INLINE void _new_thread_init(struct k_thread *thread,
+static ALWAYS_INLINE void new_thread_init(struct k_thread *thread,
```

ceolin replied:
@pfl  ? This is defined only kernel/include/kernel_structs.h. I agree with @andrewboie, this should have zephry's namespace.

pfl replied:
_new_thread_init() is only defined in ./kernel/include/kernel_structs.h and there it is defined 'static ALWAYS_INLINE'. According to the initial understanding, static functions shall only have their initial underscores removed.
You are now saying that even though this (and other, there are more similar cases) function is static, it is anyway treated as a "global" and shall therefore start with 'z_' ?
Is this special handling only applicable to ./kernel/include/kernel_structs.h or what is thinking here?

pfl replied:
Fixed.

andrewboie commented on arch/arc/include/kernel_arch_func.h:
```diff
@@ -37,7 +37,7 @@ static ALWAYS_INLINE void kernel_arch_init(void)
 
 static ALWAYS_INLINE void
-_set_thread_return_value(struct k_thread *thread, unsigned int value)
+set_thread_return_value(struct k_thread *thread, unsigned int value)
```
z_set_thread_return_value

pfl replied:
As above.

pfl replied:
Naming updated.

tyrantlizard77 reviewed (changes requested):
There's something messed up in the script which is removing leading underscores instead of changing them to z_

andrewboie commented on scripts/gen_syscall_header.py:
```diff
@@ -88,7 +88,7 @@ def gen_make_syscall(ret, argc, tabcount):
     if (ret != Retval.VOID):
         sys.stdout.write("return ")
-    sys.stdout.write("_impl_##name(")
+    sys.stdout.write("z_sys_impl_##name(")
```
why not z_impl_?

I don't like the additional _sys_ here, the names are long enough already

pfl replied:
z_sys_impl_ was taken from #9882, we can change it so something more reasonable.

pfl replied:
Updated.

pfl commented:
> There's something messed up in the script which is removing leading underscores instead of changing them to z_

Yes. It seems the code makes functions static when a certain #define is in place and global when the #define is not there. I suppose we do want to have all functions global and let the linker/compiler remove them when empty?

ceolin commented on arch/arc/core/prep_c.c:
```diff
@@ -105,7 +105,7 @@ static void adjust_vector_table_base(void)
 }
 
-extern FUNC_NORETURN void _Cstart(void);
+extern FUNC_NORETURN void z_Cstart(void);
```
I second Andy's comment to use z_cstart() and not using camelcase.

pfl replied:
Yes, fixed.

pfalcon reviewed (approved):
Ok, this is a change in the right direction apparently, so approving to get off my list. Leaving specific matters to other folks.

pfl commented:
With the initial help of the scripts in #12086 and some manual updating, this patch survives 'sanitycheck -t net' when applied on master at commit 78b644ea3a07ec6393355e01ada929066c6d50ee.

All exported fuctions should now start with the prefix 'z_', even when they are marked static.

pfl commented:
Updated to latest master.

finikorg commented on arch/arc/core/irq_manage.c:
```diff
@@ -106,7 +106,7 @@ void _irq_spurious(void *unused)
 #ifdef CONFIG_DYNAMIC_INTERRUPTS
-int _arch_irq_connect_dynamic(unsigned int irq, unsigned int priority,
+int z_arch_irq_connect_dynamic(unsigned int irq, unsigned int priority,
 			      void (*routine)(void *parameter), void *parameter,
```
is alignment now broken?

finikorg commented on arch/arc/core/mpu/arc_core_mpu.c:
```diff
@@ -104,15 +104,15 @@ void configure_mpu_mem_domain(struct k_thread *thread)
  */
-void _arch_mem_domain_partition_remove(struct k_mem_domain *domain,
+void z_arch_mem_domain_partition_remove(struct k_mem_domain *domain,
 				       u32_t partition_id)
```
and here, probably other places

pfl replied:
Done.

finikorg commented on arch/nios2/core/thread.c:
```diff
@@ -31,17 +31,17 @@ struct init_stack_frame {
 
-void _new_thread(struct k_thread *thread, k_thread_stack_t *stack,
+void z_new_thread(struct k_thread *thread, k_thread_stack_t *stack,
 		 size_t stack_size, k_thread_entry_t thread_func,
```
I think in all cases where we change `_` to `z_` alignment is broken.

finikorg reviewed (dismissed):
I think we need to correct indentation.

pfl commented:
Indentation fixed with uncrustify. Fixed a lot more than just the renamed functions.

pfl commented:
Just wondering what shippable is trying to tell me:
"geninfo: ERROR: cannot read sanity-out/native_posix/!"

pfl commented:
Hmm, `.section.text._handlermodeexit' by shippable is not touched by any patch. Is there something being generated or what should the conclusion be?

aescolar commented:
> Just wondering what shippable is trying to tell me:
"geninfo: ERROR: cannot read sanity-out/native_posix/!"

That is in the coverage report generation.
Things have failed so miserably before, as to not have generated that folder at all.

pfl committed:
all: Update reserved function names

Update reserved function names starting with one underscore, replacing
them as follows:
   '_k_' with 'z_'
   '_K_' with 'Z_'
   '_handler_' with 'z_handl_'
   '_Cstart' with 'z_cstart'
   '_Swap' with 'z_swap'

This renaming is done on both global and those static function names
in kernel/include and include/. Other static function names in kernel/
are renamed by removing the leading underscore. Other function names
not starting with any prefix listed above are renamed starting with
a 'z_' or 'Z_' prefix.

Function names starting with two or three leading underscores are not
automatcally renamed since these names will collide with the variants
with two or three leading underscores.

Various generator scripts have also been updated as well as perf,
linker and usb files. These are
   drivers/serial/uart_handlers.c
   include/linker/kobject-text.ld
   kernel/include/syscall_handler.h
   scripts/gen_kobject_list.py
   scripts/gen_syscall_header.py

Signed-off-by: Patrik Flykt <patrik.flykt@intel.com>
